### PR TITLE
Add clarity to docs on permalinks placeholders and builtins

### PR DESCRIPTION
**Pull request by EmilySeville7cfg**
## Summary

- [x] Describe `:output_ext` as a placeholder 
- [x] Add note on `W` prefix to `:week` when using `weekdate` builtin

## Questions

What to do with `date` and `pretty` formats now? I want fix documentation errors because it's hard to write JSON schema when they are exist.

### CODE REVIEW

**Comment by EmilySeville7cfg** [comment]
This is my attemp to write regex for:
- global permalink: `^((/(:(year|short_year|month|i_month|short_month|long_month|day|i_day|y_day|w_year|week|w_day|short_day|long_day|hour|minute|second|title|slug|categories|slugified_categories))+)+|date|pretty|ordinal|weekdate|none)$`
- collection permalink: `^(/(:(collection|path|name|title|output_ext))+)+$`

**Review by mattr- — APPROVED** [review]
looks good to me

**Comment by mattr-** [comment]
I went to go double check the code and now I'm confused. The documentation mirrors the code exactly. When you use these permalink styles on a site, are you seeing different results?

**Comment by EmilySeville7cfg** [comment]
1. There is an [example](https://jekyllrb.com/docs/permalinks/#global): `permalink: /:categories/:year/:month/:day/:title:output_ext`. But in [this](https://jekyllrb.com/docs/permalinks/#placeholders) placeholder list no `output_ext` is listed.

2. [Here](https://jekyllrb.com/docs/permalinks/#built-in-formats) is an example: `/:categories/:year/W:week/:short_day/:title:output_ext` but my regex can't catch `W:` and rejects this value.

**Review by ashmaroli — CHANGES_REQUESTED** [review]
Blocking this since docs should mirror Ruby codebase.

**Comment by ashmaroli** [comment]
> * There is an [example](https://jekyllrb.com/docs/permalinks/#global): `permalink: /:categories/:year/:month/:day/:title:output_ext`. But in [this](https://jekyllrb.com/docs/permalinks/#placeholders) placeholder list no `output_ext` is listed.

Agreed. But it is mentioned under [subsequent section towards end of page](https://jekyllrb.com/docs/permalinks/#collections)
Perhaps this pull request may add the missing placeholder to the global table..


> * [Here](https://jekyllrb.com/docs/permalinks/#built-in-formats) is an example: `/:categories/:year/W:week/:short_day/:title:output_ext` but my regex can't catch `W:` and rejects this value. What is `W:`? I found no information about is.

`W` will act as a prefix to placeholder `:week`.
For example, `W10`..

Agreed that it has not been documented anywhere. Perhaps this pull request could add that info..

**Comment by EmilySeville7cfg** [comment]
> W will act as a prefix to placeholder :week.

What placeholder prefixes are allowed?

**Comment by ashmaroli** [comment]
> What placeholder prefixes are allowed?

@EmilySeville7cfg See underlying code for reference:
https://github.com/jekyll/jekyll/blob/19868d5602120fe4257be1a01791be0b0686f1a5/lib/jekyll/configuration.rb#L272-L278

Built-ins are syntactic-sugars (or shorthands) to a particular implementation.
So, `weekdate` is a shorthand to a particular template: `/:categories/:year/W:week/:short_day/:title:output_ext`, i.e. no other prefix supported. If one does not want that `W`, they'll have to manually write the template without the `W`.

---

P.S. You may add a note to reduce confusion.. Also, please update the PR title and description to better describe latest diff.

File: docs/_docs/permalinks.md [hunk 1]
```diff
@@ -326,6 +334,7 @@ For posts, Jekyll also provides the following built-in styles for convenience:
       </td>
       <td>
         <p><code>/:categories/:year/W:week/:short_day/:title:output_ext</code></p>
+        <small>`W` is a prefix for `:week`</small>
```
Comment (ashmaroli) [inline docs/_docs/permalinks.md:337]:
```suggestion
        <p>
          <code>/:categories/:year/W:week/:short_day/:title:output_ext</code></br>
          <small>`W` is a prefix for `:week`</small>
        </p>
```

**Review by ashmaroli — APPROVED** [review]

**Comment by ashmaroli** [comment]
Thanks @EmilySeville7cfg
@jekyllbot: merge +doc

**Pull request merged**